tests/ml_tools/test_checkpointing.py

```diff
@@ -25,6 +27,29 @@ def dataloader(batch_size: int = 25) -> DataLoader:
     return to_dataloader(x, y, batch_size=batch_size, infinite=True)
 
 
+def write_legacy_checkpoint(
+    folder: Path,
+    model: torch.Module,
+    optimizer: Optimizer | NGOptimizer,
+    iteration: int | str,
+) -> None:
+    iteration_substring = f"{iteration:03n}" if isinstance(iteration, int) else iteration
```

Comment:
Out of curiosity, is this a torch requirement that the interation is of that format ?

Reply:
I'm not sure but I'd say it's more of a style choice rather than a pytorch requirement. My understanding is that any string can be used as a ckpt name. But @dominikandreasseitz can correct me if wrong since he wrote the original line